SwiftCompilerSources/Sources/Optimizer/PassManager/Context.swift

```diff
@@ -429,6 +429,33 @@ extension Type {
 //                          Builder initialization
 //===----------------------------------------------------------------------===//
 
+extension Instruction {
+  /// Returns self, unless self is a meta instruction, in which case the next
+  /// non-meta instruction is returned. Returns nil if there are no non-meta
+  /// instructions in the basic block.
+  fileprivate var skippingMetaInstructions: Instruction? {
+    switch self {
+    case is DebugValueInst, is AllocStackInst:
```

Comment:
I think it would make sense to add a `protocol MetaInstruction: Instruction` (ideally with a better name) and let `DebugValueInst`, `AllocStackInst` (and maybe more instructions) conform to the protocol.